src/modules/md/functions.cpp

```diff
@@ -32,3 +35,107 @@ double MDModule::determineTimeStep(const std::vector<Vec3<double>> &f)
 
     return 1.0 / fMax.absMax();
 }
+
+// Evolve Species coordinates, returning new coordinates
+std::vector<Vec3<double>> MDModule::evolve(const ProcessPool &procPool, const PotentialMap &potentialMap, const Species *sp,
+                                           double temperature, int nSteps, double deltaT,
+                                           const std::vector<Vec3<double>> &rInit, std::vector<Vec3<double>> &velocities)
+{
+    assert(sp);
+    assert(sp->nAtoms() == velocities.size());
+
+    // Create arrays
+    std::vector<double> mass(sp->nAtoms(), 0.0);
+    std::vector<Vec3<double>> forces(sp->nAtoms()), accelerations(sp->nAtoms());
+
+    // Variables
+    auto &atoms = sp->atoms();
+    double tInstant, ke, tScale;
+
+    // Units
+    // J = kg m2 s-2  -->   10 J = g Ang2 ps-2
+    // If ke is in units of [g mol-1 Angstroms2 ps-2] then must use kb in units of 10 J mol-1 K-1 (= 0.8314462)
+    const auto kb = 0.8314462;
+
+    // Store atomic masses for future use
+    for (auto &&[i, m] : zip(atoms, mass))
+        m = AtomicMass::mass(i.Z());
+
+    // Calculate total velocity and mass over all atoms
+    Vec3<double> vCom;
+    auto massSum = 0.0;
+    for (auto &&[v, m] : zip(velocities, mass))
+    {
+        vCom += v * m;
+        massSum += m;
+    }
+
+    // Remove any velocity shift
+    vCom /= massSum;
+    std::transform(velocities.begin(), velocities.end(), velocities.begin(), [vCom](auto vel) { return vel - vCom; });
+
+    // Calculate instantaneous temperature
+    ke = 0.0;
+    for (auto &&[m, v] : zip(mass, velocities))
+        ke += 0.5 * m * v.dp(v);
```

Comment:
```suggestion
    ke = std::transform_reduce(mass.begin(), mass.end(), velocities.begin(), std::plus<>(),
                               [](const auto m, const auto &v){ return 0.5 * m * v.dp(v); });
```

Again should help with vectorization